drivers/usb/usbport/roothub.c

```diff
@@ -190,6 +190,13 @@ USBPORT_RootHubClassCommand(IN PDEVICE_OBJECT FdoDevice,
                 return RHStatus;
             }
 
+            if (Port == 0 ||
+                Port >= PdoExtension->RootHubDescriptors->Descriptor.bNumberOfPorts)
```

Comment:
Naive question, related to one of my previous comments, and also because I see this sort of bound checks that sometimes use `> bNumberOfPorts`, sometimes use `>=` etc. (see https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=bNumberOfPorts ):
What's the meaning of `Port` being zero in the USB stack?

Reply:
`        RH_HubDescriptor->bNumberOfPorts = RootHubData.NumberOfPorts;`

The confusion is coming because some of these bound checks were created blindly from crappy reversing practices.
NumerOfPorts is initialized because each RootHub reports a max count
Port == ...bNumberOfPorts just means that you hit everything you can daisy chain into the same hub.

i.e your usb cart of death is  going to hit these checks over and over.

Reply:
you know what, my checks are wrong actually. 
But this conversations IS true, indeed that's how that is SUPPOSE to work. and should be updated across code base accordingly

Reply:
agreed if there's a fix for that it should be applied everywhere consistently, otherwise we would have some surprises on real hardware.